File: src/main/java/random/telegramhomebot/services/FeatureSwitcherService.java
package random.telegramhomebot.services;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import random.telegramhomebot.model.FeatureSwitcher;
import random.telegramhomebot.repository.FeatureSwitcherRepository;

import java.util.Optional;

@RequiredArgsConstructor
@Service
public class FeatureSwitcherService {

	public enum Features {NEW_HOSTS_NOTIFICATION, REACHABLE_HOSTS_NOTIFICATION, NOT_REACHABLE_HOSTS_NOTIFICATION}

	private final FeatureSwitcherRepository repository;

	public Optional<FeatureSwitcher> getFeatureSwitcherByName(String featureName) {
		return repository.findFeatureSwitcherByName(featureName);
	}

	public void saveFeatureSwitcher(FeatureSwitcher featureSwitcher) {
		repository.save(featureSwitcher);
	}

	public void switchFeature(String featureName) {
		Optional<FeatureSwitcher> featureSwitcherOptional = getFeatureSwitcherByName(featureName);
		if (featureSwitcherOptional.isPresent()) {
			FeatureSwitcher featureSwitcher = featureSwitcherOptional.get();
			featureSwitcher.setEnabled(!featureSwitcher.getEnabled());
			saveFeatureSwitcher(featureSwitcher);
		}
	}

	public boolean newHostsNotificationsEnabled() {
		Optional<FeatureSwitcher> feature = getFeatureSwitcherByName(Features.NEW_HOSTS_NOTIFICATION.name());
		return feature.isPresent() ? feature.get().getEnabled() : false;
	}

	public boolean reachableHostsNotificationsEnabled() {
		Optional<FeatureSwitcher> feature = getFeatureSwitcherByName(Features.REACHABLE_HOSTS_NOTIFICATION.name());
		return feature.isPresent() ? feature.get().getEnabled() : false;
	}

	public boolean notReachableHostsNotificationsEnabled() {
		Optional<FeatureSwitcher> feature = getFeatureSwitcherByName(Features.NOT_REACHABLE_HOSTS_NOTIFICATION.name());
		return feature.isPresent() ? feature.get().getEnabled() : false;
	}
}

File: src/main/java/random/telegramhomebot/services/FeatureSwitcherService.kt
package random.telegramhomebot.services

import org.springframework.stereotype.Service
import random.telegramhomebot.db.model.FeatureSwitcher
import random.telegramhomebot.db.repository.FeatureSwitcherRepository

@Service
class FeatureSwitcherService(private val repository: FeatureSwitcherRepository) {

    enum class Features(val command: String, val messageCode: String) {
        NEW_HOSTS_NOTIFICATION("/new_hosts_notification", "btn.newHostsNotifications"),
        REACHABLE_HOSTS_NOTIFICATION("/reachable_hosts_notification", "btn.reachableHostsNotifications"),
        NOT_REACHABLE_HOSTS_NOTIFICATION("/not_reachable_hosts_notification", "btn.notReachableNotifications")
    }

    fun switchFeature(featureName: String) {
        val featureSwitcher = getFeatureSwitcherByName(featureName)
        featureSwitcher?.let {
            it.enabled = !it.enabled
            saveFeatureSwitcher(it)
        }
    }

    fun newHostsNotificationsEnabled() = isFeatureEnabled(Features.NEW_HOSTS_NOTIFICATION)
    fun reachableHostsNotificationsEnabled() = isFeatureEnabled(Features.REACHABLE_HOSTS_NOTIFICATION)
    fun notReachableHostsNotificationsEnabled() = isFeatureEnabled(Features.NOT_REACHABLE_HOSTS_NOTIFICATION)

    private fun saveFeatureSwitcher(featureSwitcher: FeatureSwitcher) = repository.save(featureSwitcher)
    private fun getFeatureSwitcherByName(featureName: String) = repository.findFeatureSwitcherByName(featureName)
    private fun isFeatureEnabled(featureName: Features) = getFeatureSwitcherByName(featureName.name)?.enabled ?: false
}
